eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala

```diff
@@ -273,7 +274,21 @@ class Setup(val datadir: File,
           key
         }
       }
-      _ = if (bitcoinClient.useEclairSigner) logger.info("using eclair to sign bitcoin core transactions")
+      currentDescriptors <- bitcoinClient.listDescriptors()
+      _ <- if (bitcoinClient.useEclairSigner) {
+        logger.info("using eclair to sign bitcoin core transactions")
+        val descriptors = onchainKeyManager.getDescriptors(0, hSuffix = false).descriptors
+        if (!descriptors.forall(currentDescriptors.contains(_))) {
```

Comment:
If the selected wallet is not empty and already contains descriptors that do not match ours we would add them. So there may be multiple descriptors that could generate p2wpkh addresses, some will require eclair to sign txs and others will not ? It would feel safer to only import descriptors  if the wallet does not exist and has to be created first

Reply:
Good point; it makes sense to either create a new wallet and import the descriptors - or if the named wallet exists, exit with an error if the descriptors don't match.